docs/extend/admin.md

```diff
@@ -1 +1,159 @@
+
+Here's an example with a switch (boolean) item:
+
+```js
+
+app.initializers.add('interstellar', function(app) {
+
+  app.extensionData
+    .for('acme-interstellar')
+    .registerSetting(
+      {
+        setting: 'acme-interstellar.coordinates', // This is the key the settings will be saved under in the settings table in the database.
+        label: app.translator.trans('acme-interstellar.admin.coordinates_label'), // The label to be shown letting the admin know what the setting does.
+        type: 'boolean', // What type of setting this is, valid options are: boolean, text (or any other <input> tag type), and select. 
+      },
+      30 // Optional: Priority
+    )
+});
+```
+
+If you use `type: 'select'` the setting object looks a little bit different:
+
+```js
+{
+  setting: 'acme-interstellar.fuel_type',
+  label: app.translator.trans('acme-interstellar.admin.fuel_type_label'),
+  type: 'select',
+  options: {
+    'LOH': 'Liquid Fuel', // The key in this array is what the setting will be stored as in the database, the value is the label the admin will see (remember to use translations if they make sense in your context).
+    'RDX': 'Solid Fuel',
+  },
+  default: 'LOH',
+}
+```
+
+
+If you want to add something to the settings like some extra text or a more complicated input, you can also pass a callback as the first argument that returns JSX. This callback will be executed in the context of [`ExtensionPage`](https://api.docs.flarum.org/js/master/class/src/admin/components/extensionpage.js~extensionpage) and setting values will not be automatically serialized.
+
+### Registering Permissions
+
+New in beta 15, permissions can now be found in 2 places. Now, you can view each extension's individual permissions on their page. All permissions can still be found on the permissions page.
+
+In order for that to happen, permissions must be registered with `ExtensionData`. This is done in a similar way to settings, call `registerPermission`. 
+
+Arguments: 
+ * Permission object
+ * What type of permission - see [`PermissionGrid`](https://api.docs.flarum.org/js/master/class/src/admin/components/permissiongrid.js~permissiongrid)'s functions for types (remove items from the name)
+ * `ItemList` priority
+ 
+Back to our favorite rocket extension:
+
+```js
+app.initializers.add('interstellar', function(app) {
+
+  app.extensionData
+    .for('acme-interstellar')
+    .registerPermission(
+      {
+        icon: 'fas fa-rocket', // Font-Awesome Icon
+        label: app.translator.trans('acme-interstellar.admin.permissions.fly_rockets_label'), // Permission Label
+        permission: 'discussion.rocket_fly', // Actual permission name stored in database (and used when checking permission).
+      }, 
+      'start', // Category permission will be added to on the grid
+      95 // Optional: Priority
+    );
+});
+```
+
+### Chaining Reminder
+
+Remember these functions call all be chained like:
```

Comment:
call => can

Reply:
Also can you put ellipses inside of all the `register()` like `registerSetting(...)`